File: python/SendMessageOnSegment.py
"""
Ce script permet d'envoyer un message à un ségment.
"""

import requests
import Utils

def get_targets_from_segment(config, segment_id):
    """Récupération des cibles"""
    url = config['API']['url'] + 'segments/' + segment_id + '/targets/'

    # Connexion à l'API
    print("Getting all targets from segment.")
    headers = Utils.create_headers(config['API']['xKey'], 0)
    req = requests.get(url, headers=headers)
    req.raise_for_status()
    print("All targets acquired.")
    return req.json()

def send_action_to_targets(config, targets, action_id):
    """Envoie du message aux cibles"""

    print("Sending message to each target.")
    for target in targets:
        url = config['API']['url'] + "actions/" + action_id + "/targets/" + target

        # SendMessage
        headers = Utils.create_headers(config['API']['xKey'], 0)
        req = requests.post(url, headers=headers)
        req.raise_for_status()
    print("Message sent.")

def main():
    """
    Pensez à modifier les ID.
    """
    action_id = "XXXXXXX"
    segment_id = "XXXXXX"
    config = Utils.load_config()
    targets = get_targets_from_segment(config, segment_id)
    send_action_to_targets(config, targets, action_id)

if __name__ == '__main__':
    main()

File: Csharp/sendMessageOnSegment.cs
using System;
using System.IO;
using System.Net;
using System.Xml;
using Newtonsoft.Json;

namespace SendMessageOnSegment
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            // Changez le Path pour correspondre à la destination de votre fichier de configuration
            var doc = new XmlDocument();
            doc.Load("config.xml");

            var xKey = doc.DocumentElement.SelectSingleNode("/config/xKey").InnerText;
            var baseUrl = doc.DocumentElement.SelectSingleNode("/config/url").InnerText;

            const string segmentId = "01234";	//Id du segment
            const string actionId = "000000";	//Id de l'action a dupliquer

            //Ne pas remplir
            HttpWebResponse httpResponse = null;
            String responseString = null;

            //Lancement de la connexion pour remplir la requete
            String url = baseUrl + "segments/" + segmentId + "/targets/";
            Console.Write("Url : " + url + "\n");

            var result = Utils.allConnection(url, xKey, null, "GET");
            ((HttpWebResponse)result[2]).Close();
            //Verification des reponses
            if ((int)result[0] != 200)
            {
                //Affichage de l'erreur
                Console.Write("Error : {0} {1}", (int)result[0], ((HttpWebResponse)result[2]).StatusCode.ToString());
            }

            Console.Write("OK : " + (string)result[3] + "\n\n");

            var targets = JsonConvert.DeserializeObject<string[]>((string)result[3]);

            //Pour chaque Target on fait un SendMessage
            foreach (var target in targets)
            {
                //Nouvelle url
                url = baseUrl + "actions/" + actionId + "/targets/" + target;
                //SendMessage
                result = Utils.allConnection(url, xKey, null, "POST");
                ((HttpWebResponse)result[2]).Close();
                Console.Write("Url : " + url + "\n");
                if ((int)result[0] == 200)
                {
                    Console.Write("OK\n\n");
                    Console.WriteLine(target);
                }
                else
                {
                    //Affichage de l'erreur
                    Console.Write("Error : {0} {1}", (int) result[0],
                        ((HttpWebResponse) result[2]).StatusCode.ToString());
                }
            }
            Console.WriteLine("Mail sent to the segment");
            Console.ReadLine();
        }
    }
}
